webapp/lib/controller.dart

```diff
@@ -204,21 +212,37 @@ bool isProjectStale(model.NeedsReplyData projectData) {
   }
 }
 
-void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
-  projectTimers[projectData.project]?.cancel();
+String getProjectTimer(Object data) {
+  if (data is model.NeedsReplyData) {
+    return  (data as model.NeedsReplyData).project;
+  } else if(data is model.SystemMetricsData) {
+    return SYSTEM_METRICS_ROOT_COLLECTION_KEY;
+  }
+}
+
+void setupProjectTimer<T>(T projectData, [bool stale = false]) {
+  var data;
+  if (projectData is model.NeedsReplyData) {
+    data = projectData as model.NeedsReplyData;
+  } else if (projectData is model.SystemMetricsData) {
+    data = projectData as model.SystemMetricsData;
+  }
+
+  projectTimers[getProjectTimer(data)]?.cancel();
 
   if (stale) {
-    projectTimers[projectData.project] = null;
+    projectTimers[getProjectTimer(data)] = null;
   } else {
-    var timeToExecute =  projectData.datetime.add(Duration(hours: 2));
+    var timeToExecute =  data.datetime.add(Duration(hours: 2));
     var now = new DateTime.now();
     var duration = timeToExecute.difference(now);
     var timer = new Timer(duration, () {
-      if (projectData.project == selectedProject) {
-        view.contentView.stale = true;
+      if (data?.project == selectedProject) {
```

Comment:
is the null-aware `?.` operator needed here? it doesn't seem to be in the old code